File: test/before_train/testset/py-cpp/HumanEval_82_prime_length.cpp
#include<assert.h>
#include<bits/stdc++.h>
#include<vector>
#include<sstream>
#include<regex>
#include<map>
#include<set>
#include<unordered_map>
#include<unordered_set>
bool prime_length(std::string string) {
    std::vector<std::string> v;
    std::stringstream ss(string);
    std::string s;
    while (std::getline(ss, s)) {
        v.push_back(s);
    }
    std::sort(v.begin(), v.end());
    for (auto &s : v) {
        if (prime_length(s)) {
            return true;
        }
    }
    return false;
}
int main() {
    auto candidate = prime_length;
    assert(candidate(("Hello")) == (true));
    assert(candidate(("abcdcba")) == (true));
    assert(candidate(("kittens")) == (true));
    assert(candidate(("orange")) == (false));
    assert(candidate(("wow")) == (true));
    assert(candidate(("world")) == (true));
    assert(candidate(("MadaM")) == (true));
    assert(candidate(("Wow")) == (true));
    assert(candidate(("")) == (false));
    assert(candidate(("HI")) == (true));
    assert(candidate(("go")) == (true));
    assert(candidate(("gogo")) == (false));
    assert(candidate(("aaaaaaaaaaaaaaa")) == (false));
    assert(candidate(("Madam")) == (true));
    assert(candidate(("M")) == (false));
    assert(candidate(("0")) == (false));
}

File: test/before_train/testset/java-py/HumanEval_82_prime_length.py
def prime_length(string: str) -> bool:
    if (string.length() <= 1) {
        return false;
    }
    for (int i = 2; i < string.length(); i++) {
        if (string.charAt(i) == string.charAt(i - 1)) {
            return false;
        }
    }
    return true;

def check(candidate):
    assert candidate('Hello') == True
    assert candidate('abcdcba') == True
    assert candidate('kittens') == True
    assert candidate('orange') == False
    assert candidate('wow') == True
    assert candidate('world') == True
    assert candidate('MadaM') == True
    assert candidate('Wow') == True
    assert candidate('') == False
    assert candidate('HI') == True
    assert candidate('go') == True
    assert candidate('gogo') == False
    assert candidate('aaaaaaaaaaaaaaa') == False
    assert candidate('Madam') == True
    assert candidate('M') == False
    assert candidate('0') == False

def test_check():
    check(prime_length)

test_check()
